src/defaults/settings_dsl.rs

```diff
@@ -1,119 +1,198 @@
+    ($id:ident, None) => { None };
+    ($id:ident, Some($value: expr)) => { Some($value) };
+    ($id:ident, $value: expr) => { Some($value) };
+    ($id:ident, $($_: tt)*) => { ifdef![] };
 }
 
-macro_rules! add_conversion {
-    ($ctor:ident, $type:ty) => {
-        impl ToSudoDefault for $type {
-            fn to_sudodefault(self) -> SudoDefault {
-                $crate::defaults::SudoDefault::$ctor(self.into())
-            }
-        }
+macro_rules! negate_of {
+    ($id:ident, true) => {
+        false
     };
-
-    ($ctor:ident, $type:ty, negatable$(, $vetting_function:expr)?) => {
-        impl ToSudoDefault for $type {
-            #[allow(clippy::from_str_radix_10)]
-            fn to_sudodefault(self) -> SudoDefault {
-                $crate::defaults::SudoDefault::$ctor(OptTuple {
-                    default: self.into(),
-                    negated: None,
-                }$(, $vetting_function)?)
-            }
-        }
-
-        impl ToSudoDefault for ($type, $type) {
-            #[allow(clippy::from_str_radix_10)]
-            fn to_sudodefault(self) -> SudoDefault {
-                let (value, neg) = self;
-                $crate::defaults::SudoDefault::$ctor(OptTuple {
-                    default: value.into(),
-                    negated: Some(neg.into()),
-                }$(, $vetting_function)?)
-            }
-        }
+    ($id:ident, false) => {
+        false
+    };
+    ($id:ident, [ $($value: expr),* ]) => {
+        std::collections::HashSet::new()
+    };
+    ($id:ident, $(=int $check: expr;)+ $value: expr) => {
+        ifdef![]
+    };
+    ($id:ident, $(=enum $k: ident;)+ $value: ident) => {
+        ifdef![]
+    };
+    ($id:ident, None) => {
+        ifdef![]
+    };
+    ($id:ident, Some($value: expr)) => {
+        ifdef![]
+    };
+    ($id:ident, $value: expr) => {
+        ifdef![]
+    };
+    ($id:ident, $($_: tt)*) => {
+        ifdef![]
     };
 }
 
-macro_rules! sliceify {
-    ([$($value:tt),*]) => {
-        &[$($value),*][..]
+macro_rules! modifier_of {
+    ($id:ident, true) => {
+        $crate::defaults::SettingKind::Flag(Box::new(move |obj: &mut Settings| obj.$id = true))
+    };
+    ($id:ident, false) => {
+        $crate::defaults::SettingKind::Flag(Box::new(move |obj: &mut Settings| obj.$id = true))
+    };
+    ($id:ident, [ $($value: expr),* ]) => {
+        $crate::defaults::SettingKind::List(
+            |mode, list|
+                Box::new(move |obj: &mut Settings|
+                                   match mode {
+                                      ListMode::Set => obj.$id = list.into_iter().collect(),
+                                      ListMode::Add => obj.$id.extend(list),
+                                      ListMode::Del => for key in list {
+                                                           obj.$id.remove(&key);
+                                                       },
+                                   })
+        )
```

Comment:
This is how rustfmt wants to format this if you do it outside of a macro. This significantly reduces indentation.

```suggestion
        $crate::defaults::SettingKind::List(|mode, list| {
            Box::new(move |obj: &mut Settings| match mode {
                ListMode::Set => obj.$id = list.into_iter().collect(),
                ListMode::Add => obj.$id.extend(list),
                ListMode::Del => {
                    for key in list {
                        obj.$id.remove(&key);
                    }
                }
            })
        })
```

Reply:
This code is still not formatted correctly.